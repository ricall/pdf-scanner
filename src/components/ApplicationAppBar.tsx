import React from "react";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import { AppBar, Badge, Divider, IconButton, Toolbar, Typography } from "@material-ui/core";
import { Add, AppsOutlined, BookOutlined, InsertDriveFileOutlined, MenuOutlined, Save } from "@material-ui/icons";

export interface ApplicationAppBarProps {
    title?: string;
    pages?: number;
    onMenu?: () => void;
    onNewDocument?: () => void;
    onSaveDocument?: () => void;
    onEditMetadata?: () => void;
}

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        grow: {
            flexGrow: 1,
        },
        menuButton: {
            marginRight: theme.spacing(2),
        },
        title: {
            display: 'none',
            [theme.breakpoints.up('sm')]: {
                display: 'block',
            },
        },
        inputRoot: {
            color: 'inherit',
        },
        sectionDesktop: {
            display: 'none',
            [theme.breakpoints.up('md')]: {
                display: 'flex',
            },
        },
    }),
);

export const ApplicationAppBar: React.FC<ApplicationAppBarProps> = ({
    title,
    pages,
    onMenu,
    onNewDocument,
    onSaveDocument,
    onEditMetadata,
}) => {
    const classes = useStyles();

    return (
        <div className={classes.grow}>
            <AppBar position="static">
                <Toolbar>
                    <IconButton className={classes.menuButton}
                                edge="start"
                                color="inherit"

                                onClick={onMenu}>
                        <MenuOutlined/>
                    </IconButton>
                    <Typography className={classes.title} variant="h6" noWrap>{title} - pdf-scanner</Typography>
                    <div className={classes.grow}/>
                    <div className={classes.sectionDesktop}>
                        <IconButton color="inherit" onClick={onNewDocument}>
                            <Add/>
                        </IconButton>
                    </div>
                    <Divider orientation="vertical" light={false} flexItem/>
                    <div className={classes.sectionDesktop}>
                        <IconButton color="inherit" disableRipple>
                            <Badge badgeContent={pages} color="secondary">
                                <InsertDriveFileOutlined/>
                            </Badge>
                        </IconButton>
                        <IconButton color="inherit" onClick={onSaveDocument}>
                            <Badge color="secondary">
                                <Save/>
                            </Badge>
                        </IconButton>
                        <IconButton color="inherit" onClick={onSaveDocument}>
                            <Badge color="secondary">
                                <AppsOutlined/>
                            </Badge>
                        </IconButton>
                        <IconButton color="inherit" onClick={onEditMetadata}>
                            <Badge color="secondary">
                                <BookOutlined/>
                            </Badge>
                        </IconButton>
                    </div>
                </Toolbar>
            </AppBar>
        </div>
    );
}
