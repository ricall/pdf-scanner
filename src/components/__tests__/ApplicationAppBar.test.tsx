import React from 'react';
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { ApplicationAppBar } from "../ApplicationAppBar";

describe('ApplicationAppBar component', () => {

    it('can be rendered', () => {
        const wrapper = shallow(<ApplicationAppBar />);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

});
