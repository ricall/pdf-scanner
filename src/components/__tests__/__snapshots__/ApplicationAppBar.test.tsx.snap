// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationAppBar component can be rendered 1`] = `
<div
  className="makeStyles-grow-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        className="makeStyles-menuButton-2"
        color="inherit"
        edge="start"
      >
        <Memo(ForwardRef(MenuOutlinedIcon)) />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        noWrap={true}
        variant="h6"
      >
         - pdf-scanner
      </WithStyles(ForwardRef(Typography))>
      <div
        className="makeStyles-grow-1"
      />
      <div
        className="makeStyles-sectionDesktop-5"
      >
        <WithStyles(ForwardRef(IconButton))
          color="inherit"
        >
          <Memo(ForwardRef(AddIcon)) />
        </WithStyles(ForwardRef(IconButton))>
      </div>
      <WithStyles(ForwardRef(Divider))
        flexItem={true}
        light={false}
        orientation="vertical"
      />
      <div
        className="makeStyles-sectionDesktop-5"
      >
        <WithStyles(ForwardRef(IconButton))
          color="inherit"
          disableRipple={true}
        >
          <WithStyles(ForwardRef(Badge))
            color="secondary"
          >
            <Memo(ForwardRef(InsertDriveFileOutlinedIcon)) />
          </WithStyles(ForwardRef(Badge))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          color="inherit"
        >
          <WithStyles(ForwardRef(Badge))
            color="secondary"
          >
            <Memo(ForwardRef(SaveIcon)) />
          </WithStyles(ForwardRef(Badge))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          color="inherit"
        >
          <WithStyles(ForwardRef(Badge))
            color="secondary"
          >
            <Memo(ForwardRef(AppsOutlinedIcon)) />
          </WithStyles(ForwardRef(Badge))>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          color="inherit"
        >
          <WithStyles(ForwardRef(Badge))
            color="secondary"
          >
            <Memo(ForwardRef(BookOutlinedIcon)) />
          </WithStyles(ForwardRef(Badge))>
        </WithStyles(ForwardRef(IconButton))>
      </div>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
